src/VectorFiber.jl

```diff
@@ -0,0 +1,246 @@
+    return get_vector(M.manifold, M.point, X, B)
+end
+
+function get_vector!(M::TangentSpace, Y, p, X, B::AbstractBasis)
+    return get_vector!(M.manifold, Y, M.point, X, B)
+end
+
+function get_vectors(M::TangentSpace, p, B::CachedBasis)
+    return get_vectors(M.manifold, M.point, B)
+end
+
+@doc raw"""
+    injectivity_radius(M::TangentSpace)
+
+Return the injectivity radius on the [`TangentSpace`](@ref) `M`, which is $∞$.
+"""
+injectivity_radius(::TangentSpace) = Inf
+
+"""
+    inner(M::TangentSpace, p, X, Y)
+
+Inner product of vectors `X` and `Y` from the tangent space at `M`.
+"""
+function inner(M::TangentSpace, p, X, Y)
+    return inner(M.manifold, M.point, X, Y)
+end
+
+"""
+    is_flat(::TangentSpace)
+
+Return true. [`TangentSpace`](@ref) is a flat manifold.
+"""
+is_flat(::TangentSpace) = true
+
+function _isapprox(M::TangentSpace, X, Y; kwargs...)
+    return isapprox(M.manifold, M.point, X, Y; kwargs...)
+end
+
+"""
+    log(M::TangentSpace, p, q)
+
+Logarithmic map on the tangent space manifold `M`, calculated as the difference of tangent
+vectors `q` and `p` from `M`.
+"""
+log(::TangentSpace, ::Any...)
+function log!(::TangentSpace, X, p, q)
+    copyto!(X, q - p)
+    return X
+end
+
+function manifold_dimension(M::TangentSpace)
+    return manifold_dimension(M.manifold)
+end
+
+LinearAlgebra.norm(M::VectorSpaceFiber, p, X) = norm(M.manifold, M.point, X)
+
+function parallel_transport_to!(M::TangentSpace, Y, p, X, q)
+    return copyto!(M.manifold, Y, p, X)
+end
+
+@doc raw"""
+    project(M::TangentSpace, p)
+
+Project the point `p` from the tangent space `M`, that is project the vector `p`
+tangent at `M.point`.
+"""
+project(::TangentSpace, ::Any)
+
+function project!(M::TangentSpace, q, p)
+    return project!(M.manifold, q, M.point, p)
+end
+
+@doc raw"""
+    project(M::TangentSpace, p, X)
+
+Project the vector `X` from the tangent space `M`, that is project the vector `X`
+tangent at `M.point`.
+"""
+project(::TangentSpace, ::Any, ::Any)
+
+function project!(M::TangentSpace, Y, p, X)
+    return project!(M.manifold, Y, M.point, X)
+end
+
+function Random.rand!(M::TangentSpace, X; vector_at = nothing)
+    rand!(M.manifold, X; vector_at = M.point)
+    return X
+end
+function Random.rand!(rng::AbstractRNG, M::TangentSpace, X; vector_at = nothing)
+    rand!(rng, M.manifold, X; vector_at = M.point)
+    return X
+end
+
+function representation_size(B::TangentSpace)
+    return representation_size(B.manifold)
+end
+
+function Base.show(io::IO, ::MIME"text/plain", TpM::TangentSpace)
+    println(io, "Tangent space to the manifold $(base_manifold(TpM)) at point:")
+    pre = " "
+    sp = sprint(show, "text/plain", TpM.point; context = io, sizehint = 0)
+    sp = replace(sp, '\n' => "\n$(pre)")
+    return print(io, pre, sp)
+end
+
+function vector_transport_to!(M::TangentSpace, Y, p, X, q, ::AbstractVectorTransportMethod)
+    return copyto!(M.manifold, Y, p, X)
+end
+
```

Comment:
```suggestion
```


I would remove this generic definition, unless there is a good reason. This would introduce ambiguities as soon as someone wants to actually define a specific vector transport I think?

Reply:
OK, I don't exactly remember why it was needed and maybe it isn't now.